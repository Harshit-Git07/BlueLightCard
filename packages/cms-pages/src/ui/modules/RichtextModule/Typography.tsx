import { FC, ReactNode } from 'react';
import { cn } from '@/lib/utils';

const TYPOGRAPHY_STYLES = {
  normal:
    'text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body',
  h1: 'text-typography-display-small font-typography-display-small-weight leading-typography-display-small tracking-typography-display-small',
  h2: 'text-typography-headline-bold font-typography-headline-bold-weight leading-typography-headline-bold tracking-typography-headline-bold',
  h3: 'text-typography-headline font-typography-headline-weight leading-typography-headline tracking-typography-headline',
  h4: 'text-typography-title-large font-typography-title-large-weight leading-typography-title-large tracking-typography-title-large',
  h5: 'text-typography-title-medium-semibold font-typography-title-medium-semibold-weight leading-typography-title-medium-semibold tracking-typography-title-medium-semibold',
  h6: 'text-typography-title-small font-typography-title-small-weight leading-typography-title-small tracking-typography-title-small',
  blockquote:
    'border-l-2 pl-4 font-typography-body-light text-typography-body-light tracking-typography-body-light leading-typography-body-light',
};

const DEFAULT_STYLES =
  'tablet:text-typography-headline tablet:font-typography-headline-weight tablet:tracking-typography-headline tablet:leading-typography-headline';

export type TypographyProps = {
  headingLevel: 'normal' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'blockquote';
  children: ReactNode;
};

const Typography: FC<TypographyProps> = ({ headingLevel, children }) => {
  const typographyStyle = TYPOGRAPHY_STYLES[headingLevel] ?? DEFAULT_STYLES;

  const classes = cn(
    'text-heading-colour dark:text-heading-colour-dark mb-2 dark:text-colour-onSurface-dark',
    typographyStyle,
  );

  const ComponentTag = headingLevel === 'normal' ? 'p' : headingLevel;

  return <ComponentTag className={classes}>{children}</ComponentTag>;
};

export default Typography;
