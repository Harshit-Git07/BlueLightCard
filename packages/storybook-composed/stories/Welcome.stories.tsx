import type { Meta, StoryObj } from '@storybook/react';
import WelcomeComponent from './Welcome';

const meta: Meta<typeof WelcomeComponent> = {
  component: WelcomeComponent,
};

export default meta;

type Story = StoryObj<typeof WelcomeComponent>;

export const Welcome: Story = {
  args: {},
  argTypes: {},
};
