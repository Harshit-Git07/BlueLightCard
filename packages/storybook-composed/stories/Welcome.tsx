import React, { FC } from 'react';

const WelcomeComponent: FC = () => (
  <div>
    <p>
      Welcome to the Storybook Composed instance. Here you can find all of our stories across Web,
      Mobile Hybrid, and Shared UI for every brand. Please view these stories from the sidebar.
    </p>
  </div>
);

export default WelcomeComponent;
