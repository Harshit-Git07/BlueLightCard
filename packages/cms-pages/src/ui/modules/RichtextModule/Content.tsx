import { PortableText, PortableTextMarkComponentProps } from '@portabletext/react';
import Image from './Image';
import { cn } from '@/lib/utils';
import Typography from './Typography';

export default function Content({
  value,
  className,
}: { value: any } & React.HTMLProps<HTMLDivElement>) {
  return (
    <div className={cn('richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0', className)}>
      <PortableText
        value={value}
        components={{
          block: {
            normal: (node) => <Typography headingLevel="normal">{node.children}</Typography>,
            h1: (node) => <Typography headingLevel="h1">{node.children}</Typography>,
            h2: (node) => <Typography headingLevel="h2">{node.children}</Typography>,
            h3: (node) => <Typography headingLevel="h3">{node.children}</Typography>,
            h4: (node) => <Typography headingLevel="h4">{node.children}</Typography>,
            h5: (node) => <Typography headingLevel="h5">{node.children}</Typography>,
            h6: (node) => <Typography headingLevel="h6">{node.children}</Typography>,
            blockquote: (node) => (
              <Typography headingLevel="blockquote">{node.children}</Typography>
            ),
          },
          types: {
            image: Image,
            span: ({ value }) => <span>{value.text}</span>,
          },
          marks: {
            em: ({ children }: PortableTextMarkComponentProps) => <em>{children}</em>,
            strong: ({ children }: PortableTextMarkComponentProps) => <strong>{children}</strong>,
          },
        }}
      />
    </div>
  );
}
