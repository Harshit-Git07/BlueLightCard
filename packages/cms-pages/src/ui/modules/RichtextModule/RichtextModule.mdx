import { Canvas, Controls, Meta, Story, DocsContext } from '@storybook/blocks';
import ComponentStatus from '@bluelightcard/shared-ui/storybook-config/blocks/ComponentStatus';
import FigmaEmbed from '@bluelightcard/shared-ui/storybook-config/blocks/FigmaEmbed';
import * as RichtextModule from './RichtextModule.stories';

<Meta title="Richtext Module" of={RichtextModule} />

# Category

- [Status](#status)
- [Overview](#overview)
- [Props](#props)
- [State](#state)
- [Design](#design)

## Status

<ComponentStatus />

## Overview

The richtext module is used for displaying richtext content such as headings, paragraph text, blockquotes etc.

## Props

<Controls />

## State

This component does not manage any state of its own.

## Design

<FigmaEmbed />

## Variants

List of variants can be shown below here.

### Default

<Canvas of={RichtextModule.Default} />

### Heading 1

<Canvas of={RichtextModule.Heading1} />

### Heading 2

<Canvas of={RichtextModule.Heading2} />

### Heading 3

<Canvas of={RichtextModule.Heading3} />

### Heading 4

<Canvas of={RichtextModule.Heading4} />

### Heading 5

<Canvas of={RichtextModule.Heading5} />

### Heading 6

<Canvas of={RichtextModule.Heading6} />

### Blockquote

<Canvas of={RichtextModule.Blockquote} />
