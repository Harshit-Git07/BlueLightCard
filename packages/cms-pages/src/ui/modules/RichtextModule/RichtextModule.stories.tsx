import { Meta, StoryObj } from '@storybook/react';
import type { RichtextModule as RichtextModuleType } from '@bluelightcard/sanity-types';
import RichtextModule from './index';

const createTypographyMock = (
  style: 'normal' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'blockquote',
): RichtextModuleType['content'] => [
  {
    children: [
      {
        _key: `${style}-span-key`,
        _type: 'span',
        marks: [],
        text: `This is a ${style.toUpperCase()} text`,
      },
    ],
    _type: 'block',
    style,
    _key: `${style}-block-key`,
    markDefs: [],
  },
];

const normalMock = createTypographyMock('normal');
const Heading1Mock = createTypographyMock('h1');
const Heading2Mock = createTypographyMock('h2');
const Heading3Mock = createTypographyMock('h3');
const Heading4Mock = createTypographyMock('h4');
const Heading5Mock = createTypographyMock('h5');
const Heading6Mock = createTypographyMock('h6');
const BlockquoteMock = createTypographyMock('blockquote');

const meta: Meta<typeof RichtextModule> = {
  title: 'Modules/Richtext Module',
  component: RichtextModule,
  parameters: {
    status: 'done',
    design: {
      type: 'figma',
      url: 'https://www.figma.com/embed?embed_host=share&url=https://www.figma.com/design/2bUphiHBRlDyoRoctxwgGC/Static-Landing-Pages?node-id=2183-3902&t=vQH3YaYlBUUy83cT-0',
    },
  },
};

type Story = StoryObj<typeof RichtextModule>;

export const Default: Story = {
  args: {
    content: normalMock,
  },
};

export const Heading1: Story = {
  args: {
    content: Heading1Mock,
  },
};

export const Heading2: Story = {
  args: {
    content: Heading2Mock,
    tableOfContents: false,
  },
};

export const Heading3: Story = {
  args: {
    content: Heading3Mock,
    tableOfContents: false,
  },
};

export const Heading4: Story = {
  args: {
    content: Heading4Mock,
    tableOfContents: false,
  },
};

export const Heading5: Story = {
  args: {
    content: Heading5Mock,
    tableOfContents: false,
  },
};

export const Heading6: Story = {
  args: {
    content: Heading6Mock,
    tableOfContents: false,
  },
};

export const Blockquote: Story = {
  args: {
    content: BlockquoteMock,
    tableOfContents: false,
  },
};

export default meta;
