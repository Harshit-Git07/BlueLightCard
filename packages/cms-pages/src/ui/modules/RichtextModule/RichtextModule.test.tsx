import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import * as stories from './RichtextModule.stories';

const { Normal, Heading1, Heading2, Heading3, Heading4, Heading5, Heading6, Blockquote } =
  composeStories(stories);

describe('Richtext Module', () => {
  describe.each([
    { Story: Normal },
    { Story: Heading1 },
    { Story: Heading2 },
    { Story: Heading3 },
    { Story: Heading4 },
    { Story: Heading5 },
    { Story: Heading6 },
    { Story: Blockquote },
  ])('$Story.storyName', ({ Story }) => {
    it('should pass smoke test', () => {
      const result = render(<Story />);
      expect(result).toBeTruthy();
    });

    it('should match snapshot', () => {
      const result = render(<Story />);
      expect(result.container).toMatchSnapshot();
    });
  });
});
