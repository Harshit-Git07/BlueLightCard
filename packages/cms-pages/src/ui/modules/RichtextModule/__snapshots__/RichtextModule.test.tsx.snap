// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Richtext Module Blockquote should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 lg:grid-cols-[1fr,auto]"
  >
    <div
      class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0 max-w-screen-lg"
    >
      <blockquote
        class="mb-2 dark:text-colour-onSurface-dark border-l-2 pl-4 font-typography-body-light text-typography-body-light tracking-typography-body-light leading-typography-body-light"
      >
        This is a BLOCKQUOTE text
      </blockquote>
    </div>
  </section>
</div>
`;

exports[`Richtext Module Default should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 lg:grid-cols-[1fr,auto]"
  >
    <div
      class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0 max-w-screen-lg"
    >
      <p
        class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
      >
        This is a NORMAL text
      </p>
    </div>
  </section>
</div>
`;

exports[`Richtext Module Heading1 should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 lg:grid-cols-[1fr,auto]"
  >
    <div
      class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0 max-w-screen-lg"
    >
      <h1
        class="mb-2 dark:text-colour-onSurface-dark text-typography-display-small font-typography-display-small-weight leading-typography-display-small tracking-typography-display-small"
      >
        This is a H1 text
      </h1>
    </div>
  </section>
</div>
`;

exports[`Richtext Module Heading2 should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 lg:grid-cols-[1fr,auto]"
  >
    <div
      class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0 max-w-screen-lg"
    >
      <h2
        class="mb-2 dark:text-colour-onSurface-dark text-typography-headline-bold font-typography-headline-bold-weight leading-typography-headline-bold tracking-typography-headline-bold"
      >
        This is a H2 text
      </h2>
    </div>
  </section>
</div>
`;

exports[`Richtext Module Heading3 should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 lg:grid-cols-[1fr,auto]"
  >
    <div
      class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0 max-w-screen-lg"
    >
      <h3
        class="mb-2 dark:text-colour-onSurface-dark text-typography-headline font-typography-headline-weight leading-typography-headline tracking-typography-headline"
      >
        This is a H3 text
      </h3>
    </div>
  </section>
</div>
`;

exports[`Richtext Module Heading4 should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 lg:grid-cols-[1fr,auto]"
  >
    <div
      class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0 max-w-screen-lg"
    >
      <h4
        class="mb-2 dark:text-colour-onSurface-dark text-typography-title-large font-typography-title-large-weight leading-typography-title-large tracking-typography-title-large"
      >
        This is a H4 text
      </h4>
    </div>
  </section>
</div>
`;

exports[`Richtext Module Heading5 should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 lg:grid-cols-[1fr,auto]"
  >
    <div
      class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0 max-w-screen-lg"
    >
      <h5
        class="mb-2 dark:text-colour-onSurface-dark text-typography-title-medium-semibold font-typography-title-medium-semibold-weight leading-typography-title-medium-semibold tracking-typography-title-medium-semibold"
      >
        This is a H5 text
      </h5>
    </div>
  </section>
</div>
`;

exports[`Richtext Module Heading6 should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 lg:grid-cols-[1fr,auto]"
  >
    <div
      class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0 max-w-screen-lg"
    >
      <h6
        class="mb-2 dark:text-colour-onSurface-dark text-typography-title-small font-typography-title-small-weight leading-typography-title-small tracking-typography-title-small"
      >
        This is a H6 text
      </h6>
    </div>
  </section>
</div>
`;
